### PR TITLE
[BUG] fix predict function of `make reduction` (recursive, global) to work with tz aware data

### LINKED ISSUE #5463: [BUG] bug in predict when using recursive reduction with global pooling and tz aware data
**Describe the bug**

When using `make_reduction` with a recursive strategy and global pooling there is a bug in the predict function when using time-zone aware data

**To Reproduce**

```python
import numpy as np
from sktime.datasets import load_solar
from sktime.transformations.series.summarize import WindowSummarizer
from sktime.forecasting.compose import make_reduction
from sklearn.linear_model import LinearRegression

y = load_solar(
    start="2021-01-01",
    end="2022-04-01",
)
y_notz = y.copy().tz_localize(None)

window_trafo = WindowSummarizer(n_jobs=1, **{"lag_feature": {"lag": [1, 2, 48]}})
regressor = LinearRegression()
forecaster = make_reduction(
    estimator=regressor,
    strategy="recursive",
    transformers=[window_trafo],
    window_length=None,
    pooling="global",
)

# check coefficients
tzaware = forecaster.clone().fit(y).get_fitted_params()["estimator__coef"]
tznaive = forecaster.clone().fit(y_notz).get_fitted_params()["estimator__coef"]
np.all(tzaware == tznaive)

fh = np.arange(1, 97)
pred_tzaware = forecaster.clone().fit_predict(y, fh=fh)
pred_tznaive = forecaster.clone().fit_predict(y_notz, fh=fh)

# These should give us identical predictions but don't
np.all(pred_tzaware.values == pred_tznaive.values)
```

**Expected behavior**

Identical predictions

**Additional context**

**Versions**
<details>

System:
    python: 3.10.13
   machine: Windows-10-10.0.19044-SP0

Python dependencies:
          pip: 23.3
       sktime: 0.24.0
      sklearn: 1.3.1
       skbase: 0.6.0
        numpy: 1.26.1
        scipy: 1.11.3
       pandas: 2.1.1
   matplotlib: None
       joblib: 1.3.2
        numba: None
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

### PR DESCRIPTION
**Pull request by ciaran-g**
#### Reference Issues/PRs

Fixes #5463  

#### What does this implement/fix? Explain your changes.

Added
- the time zone info (if present) after the index range is calculated in `_get_shifted_window()` and `_predict_last_window()`
- a test for the bug

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### Did you add any tests for the change?

Yes

### CODE REVIEW

File: sktime/forecasting/compose/tests/test_reduce_global.py [hunk 1]
```diff
@@ -358,3 +358,30 @@ def test_nofreq_pass():
     np.testing.assert_almost_equal(
         y_pred_global["c0"].values, y_pred_nofreq["c0"].values
     )
+
+
+def test_timezoneaware_index():
+    y = load_solar(api_version=None)
+    y_notz = y.copy().tz_localize(None)
```
Comment (yarnabrina) [inline sktime/forecasting/compose/tests/test_reduce_global.py:365]: Non-blocking suggestions: `assert y.index.tz is not None` and `assert y_notz.index.tz is None`.

File: sktime/forecasting/compose/tests/test_reduce_global.py [hunk 2]
```diff
@@ -358,3 +358,30 @@ def test_nofreq_pass():
...
+    )
+
+    # check coefficients
+    tzaware = forecaster.clone().fit(y).get_fitted_params()["estimator__coef"]
+    tznaive = forecaster.clone().fit(y_notz).get_fitted_params()["estimator__coef"]
+    np.testing.assert_almost_equal(tzaware, tznaive)
+
+    fh = np.arange(1, 97)
+    pred_tzaware = forecaster.clone().fit_predict(y, fh=fh)
+    pred_tznaive = forecaster.clone().fit_predict(y_notz, fh=fh)
```
Comment (yarnabrina) [inline sktime/forecasting/compose/tests/test_reduce_global.py:384]: Non-blocking suggestions: `assert pred_tzaware.index.tz is not None` and `assert pred_tznaive.index.tz is None`.

**Review by yarnabrina — APPROVED** [review]

**Comment by fkiraly** [comment]
not blocking, but question - is this something we want to test *in general* for all estimators?

File: sktime/forecasting/compose/tests/test_reduce_global.py (hunk 1 above)
Comment (ciaran-g) [reply sktime/forecasting/compose/tests/test_reduce_global.py:365]: added 👍

File: sktime/forecasting/compose/tests/test_reduce_global.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ciaran-g) [reply sktime/forecasting/compose/tests/test_reduce_global.py]: good shout

Comment (ciaran-g) [reply sktime/forecasting/compose/tests/test_reduce_global.py]: just checked here the time zone was consistent with the input data

**Comment by ciaran-g** [comment]
> not blocking, but question - is this something we want to test _in general_ for all estimators?

I think that's a good shout tbh franz!

**Review by fkiraly — APPROVED** [review]
Thanks.

For general tests, we might want to add it to `TestAllEstimators` and see what happens.

**Pull request merged**